freesound/static/bw-frontend/src/components/modal.js

```diff
@@ -83,6 +83,31 @@ const bindConfirmationModalElements = (container) => {
   });
 }
 
+// Edit modal logic
+const bindEditModalElements = (container) => {
```

Comment:
Instead of making a new type of modal, you should be using a "generic modal with form". You have an example with the modals used to flag a sound (see https://github.com/MTG/freesound/blob/836e974274d8c00dfff855087008a0539dfa49e8/freesound/static/bw-frontend/src/pages/sound.js#L140). You should do a very similar implementation, with similar logic.